packages/nextjs/src/utils/appendDevBrowserOnCrossOrigin.ts

```diff
@@ -0,0 +1,34 @@
+// Middleware runs on the server side, before clerk-js is loaded, that's why we need Cookies.
+import type { AuthenticateRequestOptions, ClerkRequest } from '@clerk/backend/internal';
+import { constants } from '@clerk/backend/internal';
+import { DEV_BROWSER_JWT_KEY, isDevelopmentFromSecretKey, setDevBrowserJWTInURL } from '@clerk/shared';
+import { NextResponse } from 'next/server';
+
+import { SECRET_KEY } from '../server/constants';
+
+/**
+ * Grabs the dev browser JWT from cookies and appends it to the redirect URL when redirecting to cross-origin.
+ */
+export const appendDevBrowserOnCrossOrigin = (
```

Comment:
Why is this still required in the new handshake era where URL-based session syncing works for all instances?